python/dgl/transforms/functional.py

```diff
@@ -3698,15 +3722,26 @@ def laplacian_pe(g, k):
 
     # select eigenvectors with smaller eigenvalues O(n + klogk)
     EigVal, EigVec = np.linalg.eig(L.toarray())
-    kpartition_indices = np.argpartition(EigVal, k+1)[:k+1]
+    max_freqs = np.min((n-1,k))
+    kpartition_indices = np.argpartition(EigVal, max_freqs)[:max_freqs+1]
     topk_eigvals = EigVal[kpartition_indices]
     topk_indices = kpartition_indices[topk_eigvals.argsort()][1:]
     topk_EigVec = np.real(EigVec[:, topk_indices])
+    eigvals = F.tensor(np.real(EigVal[topk_indices]), dtype=F.float32)
```

Comment:
A real symmetric matrix has real eigenvalues and eigenvectors. So I think it's not necessary to use `np.real` unless the graph is not undirected.